src/types/route.ts

```diff
@@ -25,17 +21,18 @@ export type Route<
   THost extends Host = Host,
   TPath extends Path = Path,
   TQuery extends Query = Query,
-  TMeta extends RouteMeta = RouteMeta
+  TMeta extends RouteMeta = RouteMeta,
+  TStateParams extends Record<string, Param> = Record<string, Param>
 > = {
   /**
    * The specific route properties that were matched in the current route.
   */
-  matched: CreateRouteOptionsMatched & { meta: TMeta },
+  matched: CreateRouteOptions & WithHooks & (WithHost | WithoutHost) & (WithComponent | WithComponents) & (WithParent | WithoutParent) & { meta: TMeta } & { state: TStateParams },
```

Comment:
Can't this just be `CreateRouteOptionsMatched & { state: TStateParams }`? Rather than inlining the entire `CreateRouteOptionsMatched` type.